### ISSUE TITLE
Add a functionality that will close the BottomSheet when a list item is pressed

### ISSUE THREAD

**Issue by JoeDuncko** [react-native-elements/react-native-elements#2505]
**Is your feature request related to a problem? Please Describe.**

I'm trying to use the BottomSheet component to present a list of calendars to the user to add their event to. From what I can tell, there is no way to close the BottomSheet when the user has pressed one of the list items, which makes my use case difficult.

**Describe the solution you'd like**

Add a functionality that will close the BottomSheet when a list item is pressed.

As per a comment by @airen29 on https://github.com/react-native-elements/react-native-elements/pull/2325#issuecomment-671099539, it could look like this:

```javascript
onPressClose: () => {
              this.setVisible(false);
              onPress && onPress();
            },
```

**Describe alternatives you've considered**

There are some other ways this could be implemented. For example, having an `open` variable and a `onClose` function like [Material UI's modals](https://material-ui.com/components/modal/). Another option is making `cancelButtonIndex` into an array, but that doesn't seem super intuitive.

**Additional context**

**Comment by JoeDuncko** [comment]
For conversation's sake, I think it's also a bit weird how coupled the BottomSheet is to the Button component... maybe decoupling them would be good (again, like MUI's modals)...

**Comment by JoeDuncko** [comment]
I just realized that the Overlay component ( https://react-native-elements.github.io/react-native-elements/docs/overlay ) has a really similar API to what I'd like to see from the BottomSheet.

**Comment by flyingcircle** [comment]
@JoeDuncko `BottomSheet` is a very newly added component, so I think making changes to make to API more flexible is a good idea. I think you're right that there shouldn't be a button attached to the component. Instead the Modal should be controlled by the user setting `isVisible` themselves on the coponent, and thus they'd be able to manage the cancel button much more easily a customizable. 
@saxenanihal95 was the creator of the component. His input might be good as well.

**Comment by saxenanihal95** [comment]
@JoeDuncko @flyingcircle Yes i also think Button is tightly coupled to the ButtomSheet Component and also user should have a variable exposed to manage the visibility of the Modal so below the changes I am thinking to do: 

* Remove the Button Component from Bottom Sheet Component
* Expose Modal visible props and other modal props(https://reactnative.dev/docs/modal#props) to the user

**Comment by flyingcircle** [comment]
@saxenanihal95 I think those look like good changes. I think we can justify this for a 2.3 release (we'll say BottomSheet was in "beta" for 2.2) - only change missing is the docs.
@JoeDuncko does those changes look good to you or were there others that you also had in mind?

**Comment by JoeDuncko** [comment]
Looks good overall, moving conversation to the PR...

**Comment by MarcoDo99** [comment]
Hello, I am looking for the same functionality as @JoeDuncko. Is it already possible to use the new functionality? When running npm install it installs version 2.2.1 so I assume by @flyingcircle 's comment it is currently not possible?

**Comment by JoeDuncko** [comment]
@MarcoDo99 Looks like there hasn't been a release that includes #2507 yet, so no, it's not possible yet. Hopefully will be soon though!

**Comment by flyingcircle** [comment]
@MarcoDo99 @JoeDuncko this has been released in 2.3.0

**Comment by sadeedahmedkhan** [comment]
Hello, any feature or implementation for closing the BottomSheet upon pulling down?
